libyul/backends/evm/NoOutputAssembly.cpp

```diff
@@ -226,3 +236,18 @@ BuiltinFunctionForEVM const& NoOutputEVMDialect::builtin(BuiltinHandle const& _h
 		}
 	return EVMDialect::builtin(_handle);
 }
+
+std::vector<BuiltinFunctionForEVM> NoOutputEVMDialect::defineNoOutputBuiltins()
+{
+	std::vector<BuiltinFunctionForEVM> modifiedBuiltins;
+	modifiedBuiltins.reserve(allBuiltins().functions().size());
+
+	for (auto const& [_, builtin]: allBuiltins().functions())
+	{
+		auto noOutputFunction = builtin;
+		modifyBuiltinToNoOutput(noOutputFunction);
```

Comment:
What exactly does this do, or rather why? I see that it pops non-literal arguments, and then just appends zeroes for returns.

Reply:
It is used in the compilability checker where we don't actually want to create correct code but just see if it compiles, in principle. That thing is needed in the current implementation of stack compressor / stack limit evader. I think the design could be better around no output dialect and it would be good if we didn't need it at all. But at least in the current implementation we do :)